screencaptor/src/main/java/com/wealthfront/screencaptor/views/modifier/ViewDataProcessor.kt

```diff
@@ -0,0 +1,52 @@
+package com.wealthfront.screencaptor.views.modifier
+
+import android.graphics.drawable.Drawable
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+
+internal class DefaultViewDataProcessor : ViewDataProcessor {
+
+  override fun modifyViews(view: View, viewDataModifiers: Set<DataModifier>): Set<DataModifier> {
+    val listOfInitialStateModifier = mutableSetOf<DataModifier>()
+    viewDataModifiers.forEach { modifier ->
+      val initialStateModifier: DataModifier
+      when (modifier) {
+        is TextViewDataModifier -> {
+          val textView = view.findViewById<TextView>(modifier.id)
+          initialStateModifier = TextViewDataModifier(modifier.id, textView.text)
+          textView.text = modifier.data
+          listOfInitialStateModifier.add(initialStateModifier)
+        }
+        is ImageViewDataModifier -> {
+          val imageView = view.findViewById<ImageView>(modifier.id)
+          initialStateModifier = ImageViewDataModifier(modifier.id, imageView.drawable)
+          imageView.setImageDrawable(modifier.data)
+          listOfInitialStateModifier.add(initialStateModifier)
+        }
+      }
+    }
+    return listOfInitialStateModifier
+  }
+
+  override fun resetViews(view: View, initialDataModifiers: Set<DataModifier>) {
+    initialDataModifiers.forEach { modifier ->
+      when (modifier) {
+        is TextViewDataModifier -> view.findViewById<TextView>(modifier.id).text = modifier.data
+        is ImageViewDataModifier -> view.findViewById<ImageView>(modifier.id).setImageDrawable(modifier.data)
+      }
+    }
+  }
+}
+
+interface ViewDataProcessor {
+  fun modifyViews(view: View, viewDataModifiers: Set<DataModifier>): Set<DataModifier>
+  fun resetViews(view: View, initialDataModifiers: Set<DataModifier>)
+}
+
+interface DataModifier {
+  val id: Int
+}
+
+data class TextViewDataModifier(override val id: Int, val data: CharSequence) : DataModifier
+data class ImageViewDataModifier(override val id: Int, val data: Drawable) : DataModifier
```

Comment:
I'm curious about the tradeoff between keeping the actual modification code in the `ViewDataProcessor` vs keeping it in the `DataModifier`.

It might be simpler to do:
```kotlin
val myDataModifier = DataModifier<MyCustomView>(R.id.myView) { myCustomView ->
  // mutate myCustomView
}
```
And we could provide default `TextView`/`ImageViewDataModifier` that extends the default class.

Eh, after typing this out, I like the way you do it haha. Less messiness with inheritance, cleaner objects that describe the actions to be taken, and most of the time we'll be dealing with the same small set of view types anyway.

Reply:
Yeah. I though about that but here I want to restore the initial state of the views after the screenshot and the cleanest way I could do it is the way you see right now.